signing/secp256k1/api/src/main/java/tech/pegasys/signers/secp256k1/api/SingleSignerProvider.java

```diff
@@ -29,20 +30,17 @@ public SingleTransactionSignerProvider(final TransactionSigner signer) {
   }
 
   @Override
-  public Optional<TransactionSigner> getSigner(final String address) {
-    if (signer.getAddress() != null && signer.getAddress().equalsIgnoreCase(address)) {
+  public Optional<Signer> getSigner(final PublicKey publicKey) {
+    if ((signer.getPublicKey() != null)
```

Comment:
`signer.getPublicKey() != null` is performed , but not the method argument publicKey (which might result in NPE).